File: Challenges/Coding World Cup/2023/Day 2 - PAK vs NED - Unique Skills.java
import java.util.*;

public class Solution {
    static int goodSubsequence(int[] arr) {
        return (countSubseq(arr, 0, new ArrayList<>()) - 1 + 1000000007) % 1000000007;
    }

    private static int countSubseq(int[] arr, int index, List<Integer> subseq) {
        if (index == arr.length) {
            Set<Integer> uniqueSet = new HashSet<>(subseq);
            if (uniqueSet.size() == subseq.size()) {
                return 1;
            } else {
                return 0;
            }
        }
        int exclude = countSubseq(arr, index + 1, new ArrayList<>(subseq));

        subseq.add(arr[index]);
        int include = countSubseq(arr, index + 1, subseq);

        return exclude + include;
    }
}

File: Challenges/Coding World Cup/2023/Day 2 - PAK vs NED - Unique Skills.py
from typing import *


def goodSubsequence(arr: List[int]) -> int:
    # write your code here.
    def count_subseq(index, subseq):
        if index == len(arr):
            # count the number of unique elements in the subsequence
            unique_set = set(subseq)
            if len(unique_set) == len(subseq):
                return 1
            else:
                return 0

        # exclude the current element
        exclude = count_subseq(index + 1, subseq)

        # include the current element
        include = count_subseq(index + 1, subseq + [arr[index]])

        # total count is the sum of including and excluding the current element
        return exclude + include

    return (count_subseq(0, []) - 1) % 1000000007
